### PR TITLE
sourceURL returns file attachment url if file is not an image

### LINKED ISSUE #947: `sourceUrl` is false for non-image file uploads
The `sourceUrl` for PDF uploads always returns `"false"` (mimeType and other info is still in tact however) 

Image mediaItems are fine. 

Is this the right way to fetch non-image upload paths? Thanks!

Screen Shot 2019-08-22 at 10 08 23 pm

### PR DESCRIPTION
**Pull request by jydmnd**
### Your checklist for this pull request
Thanks for sending a pull request! Please make sure you click the link above to view the contribution guidelines, then fill out the blanks below.

🚨Please review the [guidelines for contributing](../CONTRIBUTING.md) to this repository.

- [ X] Make sure you are making a pull request against the **develop branch** (left side). Also you should start *your branch* off *our develop*.
- [X ] Make sure you are requesting to pull request from a **topic/feature/bugfix branch** (right side). Don't pull request from your master!

What does this implement/fix? Explain your changes.
--------------------------------------------------
The sourceURL property would return false if an uploaded file was not an image, since it was using wp_get_attachment_image_src() on non-image files.

Does this close any currently open issues?
------------------------------------------
fixes #947

Where has this been tested?
---------------------------
**Operating System:** Mac OS Mojave, CloudLinux

**WordPress Version:** 5.2.2

### CODE REVIEW

**Comment by jasonbahl** [comment]
Oooh! Thanks for getting to this. 

Any chance you'd like to contribute a test for this as well? I'd be happy to pair program with you to write the test if you're unsure about how to contribute in that way. 

Let me know. 

Thanks!

**Comment by jydmnd** [comment]
Sure! I'd love to, but I would definitely need your help.

**Comment by jydmnd** [comment]
@jasonbahl I tried installing the test suite locally, but I'm getting this error when running wpunit : 
PHP Fatal error:  Declaration of AccessFunctionsTest::setUp() must be compatible with tad\WPBrowser\Compat\Codeception\Unit::setUp(): void in [...]/wp-graphql/tests/wpunit/AccessFunctionsTest.php on line 3
I'm running PHP 7.2 on Mac OS Mojave and ran the setup script with _latest_ as the wp version, if that helps. I haven't tried with Docker yet.

It works if I add a :void at the end of the appropriate function declarations, but I'm assuming I'm not supposed to do this for every test! Maybe something from the test suite got a breaking update? Composer installed WP Browser with version 2.2.20.

As for the test itself, the bug happens when a post has a post_type of 'attachment'. I don't think there's a actual test for this specifically?

**Comment by jasonbahl** [comment]
@jydmnd if you’re available tomorrow afternoon (MST) we could do a pairing session and get the tests running and get some written for this. 

Or we can schedule something next week. 

What time zone are you in?

**Comment by jydmnd** [comment]
@jasonbahl I'm available tomorrow from 1PM to 5PM EST.

**Comment by jasonbahl** [comment]
@jydmnd are you in the WPGraphQL Slack? If not you can join here: https://wpgql-slack.herokuapp.com/

If you can DM me in there we can coordinate a Zoom call to pair program. 

Thanks!

File: src/Model/Post.php [hunk 1]
```diff
@@ -408,10 +408,10 @@ protected function init() {
 					'mediaType'           => function() {
 						return wp_attachment_is_image( $this->data->ID ) ? 'image' : 'file';
 					},
+                    'mediaItemUrl' => function() {
```
Comment (jasonbahl) [inline src/Model/Post.php]: spacing looks off here

File: src/Type/Object/PostObject.php [hunk 2]
```diff
@@ -164,13 +164,17 @@ function register_post_object_types( $post_type_object ) {
 					'type'        => 'String',
 					'description' => __( 'Type of resource', 'wp-graphql' ),
 				],
+                'mediaItemUrl'    => [
+                    'type'        => 'String',
+                    'description' => __( 'Url of the mediaItem', 'wp-graphql' ),
+                ],
```
Comment (jasonbahl) [inline src/Type/Object/PostObject.php]: spacing looks off here

File: tests/wpunit/MediaItemQueriesTest.php [hunk 3]
```diff
@@ -336,4 +336,37 @@ public function testMediaItemQuery( $image_meta = [] ) {
...
 	}
 
+    /**
+     * testPostQuery
+     *
+     * This tests creates a single attachment and retrieves said post URL via a GraphQL query
+     *
+     * @since 0.3.6
+     */
+    public function testMediaItemUrl() {
```
Comment (jasonbahl) [inline tests/wpunit/MediaItemQueriesTest.php]: This is good! 

We should include another test to test something of another mime type, like a video or mp3 or something. 

I think the common complaint has been that `sourceUrl` wasn't working for non-images

**Comment by jasonbahl** [comment]
@jydmnd there was a few formatting issues, so I just pushed up some quick changes. I think this is looking pretty good. I want to verify a few things locally though before merging.

**Comment by jasonbahl** [comment]
Ok, so I wonder if it makes sense to deprecate the `sourceUrl` field here. Let it continue resolving, but mark it deprecated so that it no longer shows in GraphiQL Docs. 🤔

**Comment by jydmnd** [comment]
@jasonbahl I don't think it should be deprecated, since sourceUrl can still be used for pictures when providing a size. It returns a different URL depending on the size, which is fine.

**Comment by jasonbahl** [comment]
@jydmnd ah ya. 🤔 

I think this feature will need to be revisited with a more critical eye again in the future, but for now I think the value of this is high enough for users to merge as-is. 

I think down the road we'll need to formally address a MediaItem interface Type and have separate types for each Type of Media Item that support different shapes. 

Will need to think about it a bit more and probably work on an RFC.

**Comment by jasonbahl** [comment]
closes #947

**Pull request merged**